src/drain.rs

```diff
@@ -12,7 +12,7 @@ use never::Never;
 /// when a drain is signaled.
 pub fn channel() -> (Signal, Watch) {
     let (tx, rx) = oneshot::channel();
-    let (drained_tx, drained_rx) = mpsc::channel(0);
+    let (drained_tx, drained_rx) = mpsc::channel(1);
```

Comment:
Will be interesting to know whats the significance of a zero sized mpsc::channel. Tokio does not allow that. Would a chan of size one be ok here ?

Reply:
The futures channel always had 1 buffer space per sender, so even 0 realistically meant 1. Tokio-sync doesn't add buffer space per sender.

This channel is never actually sent on, it's just used since senders can be cloned, and the receiver can be notified when the last clone drops.

Reply:
Ah, I see. Thanks for the clarification